src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/utils/PsiTypeUtils.java

```diff
@@ -178,6 +185,34 @@ public static String getOptionalTypeParameter(String typeName) {
         return typeName.substring(start + 1, end);
     }
 
+    /**
+     * Returns the first type parameter for the given <code>psiType</code>.
+     * Returns the given <code>psiType</code> if the given psiType has no generic type parameter.
+     *
+     * @param psiType the psiType to return the generic type parameter for
+     * @return the generic type parameter for the given psiType
+     */
+    public static PsiType getTypeParameter(PsiType psiType) {
+        PsiType type = psiType;
```

Comment:
@angelozerr unsure of this default: if the given type has no generic parameter (ex. OptionalInt) I return the type as is.

Reply:
@angelozerr would love to get your feedback on this since you did the original.